shopfloor/services/checkout.py

```diff
@@ -564,35 +564,37 @@ def _select_lines_from_lot(
                     "body": _("Lot is not in the current transfer."),
                 },
             )
-
-        # When lots are as units outside of packages, we can select them for
-        # packing, but if they are in a package, we want the user to scan the packages.
-        # If the product is only in one package though, scanning the lot selects
-        # the package.
         packages = lines.mapped("package_id")
         # Do not use mapped here: we want to see if we have more than one
         # package, but also if we have one lot as a package and the same lot as
         # a unit in another line. In both cases, we want the user to scan the
         # package.
         if packages and len({line.package_id for line in lines}) > 1:
+            # The lot is spread out in multiple packages, ask to scan a package
             return self._response_for_select_line(
                 picking, message=self.msg_store.lot_multiple_packages_scan_package()
             )
         elif packages:
-            # Select all the lines of the package when we scan a lot in a
-            # package and we have only one.
-            return self._select_lines_from_package(
-                picking, selection_lines, packages, prefill_qty=prefill_qty, **kw
+            # The lot is only in one package
+            # Related lines are the one in the same package
+            related_lines = selection_lines.filtered(
```

Comment:
I don't know what means "related lines" here compared to "line_to_update" or "lines" below. Can you give a definition in a comment?

Reply:
So if the related lines are the lines belonging to the same package than the lot line, we want to scan a package right? I don't know where this is happening.